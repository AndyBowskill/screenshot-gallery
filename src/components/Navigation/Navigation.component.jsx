import React from 'react';
import './Navigation.style.css';

const Navigation = ({ onRouteChange, isUserSignedIn }) => {
  if (isUserSignedIn) {
    return (
      <nav className='navigation'>
        <p className='navigation-logo'>Screenshot Gallery</p>
        <ul className='navigation-links'>
          <li onClick={() => onRouteChange('signin')}>Sign Out</li>
        </ul>
      </nav>
    );
  } else {
    return (
      <nav className='navigation'>
        <p className='navigation-logo'>Screenshot Gallery</p>
        <ul className='navigation-links'>
          <li onClick={() => onRouteChange('signin')}>Sign In</li>
          <li onClick={() => onRouteChange('register')}>Register</li>
        </ul>
      </nav>
    );
  }
};

export default Navigation;
