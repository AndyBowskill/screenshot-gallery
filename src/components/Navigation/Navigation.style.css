.navigation {
    display: flex;
    justify-content: flex-end;
    font-size: 1.2rem;
    background-color: #0074D9;
    color: rgb(127, 219, 255);
    padding: 20px 20px;
    border-bottom: 2px solid #eee;
}