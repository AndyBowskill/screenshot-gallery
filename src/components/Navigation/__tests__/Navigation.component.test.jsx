import React from 'react';
import { render, screen } from '@testing-library/react';
import UserEvent from '@testing-library/user-event';
import renderer from 'react-test-renderer';
import Navigation from '../Navigation.component';

test('Navigation component displays correctly when the user is signed out.', () => {
  const onRouteChangeFn = jest.fn();

  const component = renderer.create(
    <Navigation onRouteChange={onRouteChangeFn} isUserSignedIn={false} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Navigation component displays correctly when the user is signed in.', () => {
  const onRouteChangeFn = jest.fn();

  const component = renderer.create(
    <Navigation onRouteChange={onRouteChangeFn} isUserSignedIn={true} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Elements display correctly when the user is signed out.', () => {
  const onRouteChangeFn = jest.fn();

  render(<Navigation onRouteChange={onRouteChangeFn} isUserSignedIn={true} />);

  const signOutElement = screen.getAllByText(/Sign Out/);
  UserEvent.click(signOutElement[0]);
  expect(signOutElement[0]).toBeInTheDocument();
});

test('Elements display correctly when the user is signed out.', () => {
  const onRouteChangeFn = jest.fn();

  render(<Navigation onRouteChange={onRouteChangeFn} isUserSignedIn={false} />);

  const signInElement = screen.getAllByText(/Sign In/);
  UserEvent.click(signInElement[0]);
  expect(signInElement[0]).toBeInTheDocument();

  const registerElement = screen.getAllByText(/Register/);
  UserEvent.click(registerElement[0]);
  expect(registerElement[0]).toBeInTheDocument();
});
