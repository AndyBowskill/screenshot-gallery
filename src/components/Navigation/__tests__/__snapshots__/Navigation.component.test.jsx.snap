// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation component displays correctly when the user is signed in. 1`] = `
<nav
  className="navigation"
>
  <p
    className="navigation-logo"
  >
    Screenshot Gallery
  </p>
  <ul
    className="navigation-links"
  >
    <li
      onClick={[Function]}
    >
      Sign Out
    </li>
  </ul>
</nav>
`;

exports[`Navigation component displays correctly when the user is signed out. 1`] = `
<nav
  className="navigation"
>
  <p
    className="navigation-logo"
  >
    Screenshot Gallery
  </p>
  <ul
    className="navigation-links"
  >
    <li
      onClick={[Function]}
    >
      Sign In
    </li>
    <li
      onClick={[Function]}
    >
      Register
    </li>
  </ul>
</nav>
`;
